backends/ref/ceed-ref-restriction.c

```diff
@@ -79,8 +79,14 @@ static inline int CeedElemRestrictionApply_Ref_Core(CeedElemRestriction r, const
       for (CeedInt e = start * blk_size; e < stop * blk_size; e += blk_size) {
         CeedPragmaSIMD for (CeedInt k = 0; k < num_comp; k++) {
           CeedPragmaSIMD for (CeedInt i = 0; i < elem_size * blk_size; i++) {
-            vv[elem_size * (k * blk_size + num_comp * e) + i - v_offset] =
-                uu[impl->offsets[i + elem_size * e] + k * comp_stride] * (is_oriented && impl->orient[i + elem_size * e] ? -1. : 1.);
+            if (!use_orient || !impl->orient) {
+              // Unsigned restriction
+              vv[elem_size * (k * blk_size + num_comp * e) + i - v_offset] = uu[impl->offsets[i + elem_size * e] + k * comp_stride];
+            } else {
+              // Signed restriction
+              vv[elem_size * (k * blk_size + num_comp * e) + i - v_offset] =
+                  uu[impl->offsets[i + elem_size * e] + k * comp_stride] * (impl->orient[i + elem_size * e] ? -1.0 : 1.0);
+            }
```

Comment:
We should double check for a performance impact here. This tends to be an operation that we spend a fair amount of time in and I don't think we've dug into making sure this code builds the best assembly it can.

Reply:
Here's a small example which (if I am reading things correctly) shows this test being optimized away around optimization -O2: https://godbolt.org/z/9cPfEGqzr.

Of course I can run benchmarks too. Are there any in particular you'd like me to run?

Reply:
Thanks for so thoroughly checking - looks like any difference is negligible to me